crates/trippy/src/main.rs

```diff
@@ -485,20 +485,39 @@ mod tests {
     use super::*;
     use test_case::test_case;
 
-    #[test_case(include_str!("../test_resources/tui/tui_theme_items.txt"), &tui_theme_items(); "tui theme items match")]
-    #[test_case(include_str!("../test_resources/tui/tui_binding_commands.txt"), &tui_binding_commands(); "tui binding commands match")]
-    #[test_case(include_str!("../test_resources/config/completions_bash.txt"), &shell_completions(Shell::Bash).unwrap(); "generate bash shell completions")]
-    #[test_case(include_str!("../test_resources/config/completions_elvish.txt"), &shell_completions(Shell::Elvish).unwrap(); "generate elvish shell completions")]
-    #[test_case(include_str!("../test_resources/config/completions_fish.txt"), &shell_completions(Shell::Fish).unwrap(); "generate fish shell completions")]
-    #[test_case(include_str!("../test_resources/config/completions_powershell.txt"), &shell_completions(Shell::PowerShell).unwrap(); "generate powershell shell completions")]
-    #[test_case(include_str!("../test_resources/config/completions_zsh.txt"), &shell_completions(Shell::Zsh).unwrap(); "generate zsh shell completions")]
-    #[test_case(include_str!("../test_resources/config/trip.1"), &man_page().unwrap(); "generate man page")]
-    fn test_output(expected: &str, actual: &str) {
-        if remove_whitespace(actual.to_string()) != remove_whitespace(expected.to_string()) {
-            pretty_assertions::assert_eq!(actual, expected);
+    #[macro_export]
+    macro_rules! set_snapshot_settings {
+        ($info:expr, $($expr:expr),*) => {
+            let mut settings = insta::Settings::clone_current();
+
+            // Set raw_text save in the snapshot (unprocessed version of output can go here)
+            settings.set_info(&format!("{:#?}",$info));
+
+            // Set snapshot suffix
+            settings.set_snapshot_suffix(format!($($expr,)*));
+
+            // Set snapshot storage location
+            let cwd = std::env::current_dir().expect("failed to get cwd");
+            let path = cwd.join("tests").join("resources").join("snapshots");
+            settings.set_snapshot_path(path);
+
+            let _guard = settings.bind_to_scope();
```

Comment:
Could we use `insta::with_settings!` for this?  It would mean passing the test body to the macro as an argument but I don't think that would be an issue?  Again, not a deal breaker.

Reply:
Previously before I saw the example macro on their website I used to actually just have a [function](https://github.com/rust-practice/cargo-leet/blob/5f02af149192f151afaad7d288ec80e86c503b53/src/tool/core/helpers/local_store.rs#L59) that returned the settings. But I found the macro more ergonomic, but that's pretty subjective so I'm ok with going with what you prefer. Would you prefer not using a macro?

Reply:
Well you said it would be nice so does that mean you're prefer not use the macro (Sorry just wanted to clarify what you want)?

Reply:
Changed it to have an `insta` fn which takes a closure and removed the macro